app/src/main/cpp/skyline/services/sm/IUserInterface.cpp

```diff
@@ -4,26 +4,28 @@
 #include "IUserInterface.h"
 
 namespace skyline::service::sm {
-    IUserInterface::IUserInterface(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager, Service::sm_IUserInterface, "sm:IUserInterface", {
+    IUserInterface::IUserInterface(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager, {
         {0x0, SFUNC(IUserInterface::Initialize)},
         {0x1, SFUNC(IUserInterface::GetService)}
     }) {}
 
-    void IUserInterface::Initialize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {}
+    Result IUserInterface::Initialize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return {};
+    }
+
+    Result IUserInterface::GetService(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto name = request.Pop<ServiceName>();
 
-    void IUserInterface::GetService(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        std::string serviceName(request.PopString());
-        serviceName.resize(std::min(8UL, serviceName.size()));
+        if (!name)
+            return result::InvalidServiceName;
 
-        if (serviceName.empty()) {
-            response.errorCode = constant::status::ServiceInvName;
-        } else {
-            try {
-                manager.NewService(serviceName, session, response);
-            } catch (std::out_of_range &) {
-                response.errorCode = constant::status::ServiceNotReg;
-                state.logger->Warn("Service has not been implemented: \"{}\"", serviceName);
-            }
+        try {
+            manager.NewService(name, session, response);
+        } catch (std::out_of_range &) {
+            std::string_view stringName(reinterpret_cast<char *>(&name), sizeof(u64));
+            state.logger->Warn("Service has not been implemented: \"{}\"", stringName);
```

Comment:
I wonder, should we return `result::InvalidServiceName` here or not?

Reply:
That would be the entirely wrong error code, so no